src/blueprint/azext_blueprint/_params.py

```diff
@@ -0,0 +1,215 @@
+        c.argument('artifact_name', help='Name of the blueprint artifact.')
+        c.argument('display_name', help='DisplayName of this artifact.')
+        c.argument('description', help='Description of the blueprint artifact.')
+        c.argument('depends_on', nargs='+', help='Artifacts which need to be deployed before the specified artifact.')
+        c.argument('role_definition_id', help='Azure resource ID of the RoleDefinition.')
+        c.argument('resource_group_art', help='Name of the resource group artifact to which the policy will be assigned.')
+        c.argument('principal_ids', help='Array of user or group identities in Azure Active Directory. The roleDefinition will apply to each identity.')
+
+    with self.argument_context('blueprint artifact role update') as c:
+        c.argument('blueprint_name', help='Name of the blueprint definition.')
+        c.argument('artifact_name', help='Name of the blueprint artifact.')
+        c.argument('display_name', help='DisplayName of this artifact.')
+        c.argument('description', help='Description of the blueprint artifact.')
+        c.argument('depends_on', nargs='+', help='Artifacts which need to be deployed before the specified artifact.')
+        c.argument('resource_group_art', help='Name of the resource group artifact to which the policy will be assigned.')
+
+    with self.argument_context('blueprint artifact template create') as c:
+        c.argument('blueprint_name', help='Name of the blueprint definition.')
+        c.argument('artifact_name', help='Name of the blueprint artifact.')
+        c.argument('display_name', help='DisplayName of this artifact.')
+        c.argument('description', help='Description of the blueprint artifact.')
+        c.argument('depends_on', nargs='+', help='Artifacts which need to be deployed before the specified artifact.')
+        c.argument('resource_group_art', help='Name of the resource group artifact to which the policy will be assigned.')
+        c.argument('parameters', arg_type=parameter_type)
+        c.argument('template', arg_type=template_type)
+
+    with self.argument_context('blueprint artifact template update') as c:
+        c.argument('blueprint_name', help='Name of the blueprint definition.')
+        c.argument('artifact_name', help='Name of the blueprint artifact.')
+        c.argument('display_name', help='DisplayName of this artifact.')
+        c.argument('description', help='Description of the blueprint artifact.')
+        c.argument('depends_on', nargs='+', help='Artifacts which need to be deployed before the specified artifact.')
+        c.argument('resource_group_art', help='Name of the resource group artifact to which the policy will be assigned.')
+        c.argument('parameters', arg_type=parameter_type)
+        c.argument('template', arg_type=template_type)
+
+    with self.argument_context('blueprint published create') as c:

```

Comment:
`blueprint publish create`? we usually use noun for subgroup name.

Reply:
As required by PM, use `blueprint publish` directly.